compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -2297,6 +2304,99 @@ public BLangNode transform(ArrayTypeDescriptorNode arrayTypeDescriptorNode) {
         return arrayTypeNode;
     }
 
+    public void addEnumDeclarationNode(
+            EnumDeclarationNode enumDeclarationNode,
+            BLangCompilationUnit compInit) {
+
+        Boolean publicQualifier = false;
+        if (enumDeclarationNode.qualifier() != null &&
+                enumDeclarationNode.qualifier().kind()
+                        == SyntaxKind.PUBLIC_KEYWORD) {
+            publicQualifier = true;
+        }
+        for (Node member : enumDeclarationNode.enumMemberList()) {
+            compInit.addTopLevelNode(transformEnumMember(
```

Comment:
please use `addToTop` function instead. because `compilationUnit` can be `null` in lang-server

Reply:
alright :)